hooks/pre/20-certs_update.rb

```diff
@@ -0,0 +1,66 @@
+require 'fileutils'
+require 'English'
+
+if module_enabled?('katello')
+  SSL_BUILD_DIR = param('certs', 'ssl_build_dir').value
+  CHECK_SCRIPT = `which katello-certs-check`.strip
+  
+  def mark_for_update(cert_name, hostname = nil)
+    path = File.join(*[SSL_BUILD_DIR, hostname, cert_name].compact)
+    if app_value(:noop)
+      puts "Marking certificate #{path} for update (noop)"
+    else
+      puts "Marking certificate #{path} for update"
+      FileUtils.touch("#{path}.update")
+    end
+  end
+
+  ca_file   = param('certs', 'server_ca_cert').value
+  cert_file = param('certs', 'server_cert').value
+  key_file  = param('certs', 'server_key').value
+  
+  if app_value('certs_update_server_ca') && !module_enabled?('katello')
+    fail_and_exit("--certs-update-server-ca needs to be used with katello", 101)
+  end
+  
+  if param('foreman_proxy_certs', 'foreman_proxy_fqdn')
+    hostname = param('foreman_proxy_certs', 'foreman_proxy_fqdn').value
+  else
+    hostname = param('certs', 'node_fqdn').value
+  end
+  
+  if app_value('certs_update_server')
+    mark_for_update("#{hostname}-apache", hostname)
+    mark_for_update("#{hostname}-foreman-proxy", hostname)
+  end
+  
+  if app_value('certs_update_all') || app_value('certs_update_default_ca') || app_value('certs_reset')
+    all_cert_names = Dir.glob(File.join(SSL_BUILD_DIR, hostname, '*.noarch.rpm')).map do |rpm|
+      File.basename(rpm).sub(/-1\.0-\d+\.noarch\.rpm/, '')
+    end.uniq
+  
+    all_cert_names.each do |cert_name|
+      mark_for_update(cert_name, hostname)
+    end
+  end
+  
+  if app_value('certs_update_server_ca') || app_value('certs_reset')
+    mark_for_update('katello-server-ca')
+  end
+  
+  if !app_value('certs_skip_check') &&
+     cert_file.to_s != "" &&
+     (app_value('certs_update_server_ca') || app_value('certs_update_server'))
+    check_cmd = %(#{CHECK_SCRIPT} -c "#{cert_file}" -k "#{key_file}" -b "#{ca_file}")
+    output = `#{check_cmd} 2>&1`
+    unless $CHILD_STATUS.success?
+      fail_and_exit("Command '#{check_cmd}' exited with #{$CHILD_STATUS.exitstatus}:\n #{output}", 101)
+    end
```

Comment:
We have `execute` wrappers for this now. Should we use those instead?

Reply:
```suggestion
    check_cmd = %(katello-certs-check -c "#{cert_file}" -k "#{key_file}" -b "#{ca_file}")
    unless execute(check_cmd)
      fail_and_exit("Command '#{check_cmd}' failed, exiting.", 101)
    end
```

I think we could so something like this. `execute(check_cmd)` would still log and print output including any errors but I don't see how we could then pass any exit code or error output to `fail_and_exit` -- which I guess is not necessary at that point because `execute` is printing and logging it already?